src/containers/MyNdla/components/Folder.tsx

```diff
@@ -0,0 +1,236 @@
+    left: 0;
+  }
+`;
+
+const FolderTitle = styled.h2`
+  ${fonts.sizes("16px", "20px")};
+  font-weight: ${fonts.weight.semibold};
+  margin: 0px !important;
+  flex: 1;
+
+  overflow: hidden;
+  text-overflow: ellipsis;
+  /* Unfortunate css needed for multi-line text overflow ellipsis. */
+  display: -webkit-box;
+  -webkit-line-clamp: 1;
+  line-clamp: 1;
+  -webkit-box-orient: vertical;
+`;
+
+const MenuWrapper = styled.div`
+  display: flex;
+  z-index: ${stackOrder.offsetSingle};
+  flex-direction: row;
+  align-items: center;
+  justify-content: space-between;
+`;
+
+const CountContainer = styled.div`
+  overflow: hidden;
+  display: flex;
+  flex-direction: row;
+  min-height: 44px;
+  gap: ${spacing.small};
+  margin: 0 ${spacing.small} 0 ${spacing.nsmall};
+
+  ${mq.range({ until: breakpoints.tablet })} {
+    &[data-type="list"] {
+      display: none;
+    }
+  }
+`;
+
+const IconTextWrapper = styled.div`
+  display: flex;
+  align-items: center;
+  gap: ${spacing.xxsmall};
+  color: ${colors.brand.grey};
+  white-space: nowrap;
+  svg {
+    width: 13px;
+    height: 13px;
+  }
+  ${fonts.sizes(16)};
+`;
+
+interface IconCountProps {
+  type: "resource" | "folder";
+  count?: number;
+  layoutType: LayoutType;
+}
+
+const Count = ({ type, count, layoutType }: IconCountProps) => {
+  const Icon = type === "resource" ? FileDocumentOutline : FolderOutlined;
+  const { t } = useTranslation();
+  if (!count) return null;
+
+  return (
+    <IconTextWrapper data-type={layoutType}>
+      <Icon />
+      <span>{t(`myNdla.${type}s`, { count })}</span>
+    </IconTextWrapper>
+  );
+};
+
+interface Props {
+  description?: string;
+  type?: LayoutType;
+  menu?: ReactNode;
+  folder: GQLFolder;
+  foldersCount: Record<string, FolderTotalCount>;
+}
+
+const getIcon = (isOwner: boolean, isShared?: boolean) => {
+  if (!isOwner) {
+    return Link;
+  } else if (isShared) {
+    return FolderSharedOutlined;
+  } else {
+    return FolderOutlined;
+  }
+};
+
+export const Folder = ({
+  type = "list",
+  menu,
+  folder: { id, status, __typename, name, owner },
+  foldersCount,
+}: Props) => {
+  const { t } = useTranslation();
+  const Icon = getIcon(__typename === "Folder", status === "shared");
+
+  return (
+    <FolderWrapper data-type={type} id={id}>
+      <TitleWrapper data-type={type}>
+        <IconWrapper
+          aria-label={`${status === "shared" ? `${t("myNdla.folder.sharing.shared")} ` : ""}${t("myNdla.folder.folder")}`}
+        >
+          <Icon />
+        </IconWrapper>
+        <ResourceTitleLink to={__typename === "Folder" ? routes.myNdla.folder(id) : routes.folder(id)}>
+          <FolderTitle data-title="" title={name}>
+            {name}
+          </FolderTitle>
+        </ResourceTitleLink>
+      </TitleWrapper>
+      <MenuWrapper>
+        <CountContainer data-type={type}>
+          {status === "shared" && (
+            <IconTextWrapper>
+              <Share />
+              {__typename !== "Folder" ? (
```

Comment:
Det er veldig masse sjekk på __type her. Det forteller meg at du forsøker å gjøre to ting med en komponent. Då er det bedre at du heller gjør folder-komponenten enklere og sender inn children som parameter. Du kunne feks sendt inn ikon, link og children inn.

Reply:
Fremdeles masse sjekk på __typename. kvar er isFolder som vi foreslo?